test/e2e/serial/tests/configuration.go

```diff
@@ -201,8 +203,9 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 
 				testMcp := objects.TestMCP()
 				By(fmt.Sprintf("deleting mcp: %q", testMcp.Name))
-				err = fxt.Client.Delete(context.TODO(), testMcp)
-				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					return fxt.Client.Delete(context.TODO(), testMcp)
```

Comment:
same goes for delete:

1. the deletion request is accepted/rejected synchronously
2. then we need to wait for the object (s) to be gone, and if we don't do it, that's the bug we need to solve